File: android/app/src/main/java/com/adqmobile/socialnetwork/domain/user/usecases/GetAllUsersUseCase.kt
package com.adqmobile.socialnetwork.domain.user.usecases

import com.adqmobile.socialnetwork.domain.user.User
import com.adqmobile.socialnetwork.domain.user.UserRepository

class GetAllUsersUseCase(private val userRepository: UserRepository) {

    suspend fun run(): Array<User> {
        return userRepository.retrieveAll()
    }
}

File: flutter/lib/domain/user/usecases/getAllUsersUseCase.dart
import '../userRepository.dart';

import '../user.dart';

class GetAllUsersUseCase {
  UserRepository userRepository;

  GetAllUsersUseCase(UserRepository userRepository) {
    this.userRepository = userRepository;
  }

  Future<List<User>> run() {
    return userRepository.retrieveAll();
  }
}